### PR TITLE
PE-878: validator for the file name

### PR DESCRIPTION
**Pull request by matibat**
https://ardrive.atlassian.net/wiki/spaces/ENGINEERIN/pages/313950220/SPIKE+ArFS+Standards

### CODE REVIEW

**Review by jdaev — APPROVED** [review]
Looks good 👍🏽

**Review by fedellen — APPROVED** [review]
Looking good!

File: src/arfs/input_validators.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+/* RegExp explanation
+ * - `^$` matches the beginning and the end of the string
+ * - for each character, we ensure it is not a reserved one, and
+ *   - that it's not preceded by a space at the beginning
+ *   - nor a space or point at the end
+ */
+const VALID_FILE_NAME_REGEXP = new RegExp(`^(?:(?!^\\s)[^${RESERVED_CHARACTERS.join('')}](?![. ]$))+$`);
+const MAX_VALID_NAME_LENGTH = 255;
+
+export function assertValidName(name: string): boolean {
```
Comment (arielmelendez) [inline src/arfs/input_validators.ts:11]: Let's be more specific about the type of name we're enforcing here, i.e. `ArFsEntityName`

File: src/arfs/input_validators.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const RESERVED_CHARACTERS = ['\\\\', '/', ':', '*', '?', '"', '<', '>', '|'];
+/* RegExp explanation
+ * - `^$` matches the beginning and the end of the string
+ * - for each character, we ensure it is not a reserved one, and
+ *   - that it's not preceded by a space at the beginning
+ *   - nor a space or point at the end
+ */
+const VALID_FILE_NAME_REGEXP = new RegExp(`^(?:(?!^\\s)[^${RESERVED_CHARACTERS.join('')}](?![. ]$))+$`);
```
Comment (arielmelendez) [inline src/arfs/input_validators.ts:8]: If we're going to use a non-capturing group and negative lookahead, let's explain why.

File: src/arfs/input_validators.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			'::::::::::.png',
+			'**********.png',
+			'??????????.png',
+			'"""""""""".png',
+			'<<<<<<<<<<.png',
+			'>>>>>>>>>>.png',
+			'||||||||||.png'
+		];
+
+		it('returns true when fed with an ArFS compliant name', () => {
```
Comment (arielmelendez) [inline src/arfs/input_validators.test.ts:26]: Missing test cases for starting or ending with spaces

Comment (arielmelendez) [reply src/arfs/input_validators.test.ts:26]: Some test cases I was using in rubular.com:
```
 this is invalid
alsoinvalid 
not*valid
notvalid.
```

Comment (matibat) [reply src/arfs/input_validators.test.ts]: The missing test cases added: 89d8f54

File: src/arfs/input_validators.ts (hunk 2 above)
Comment (matibat) [reply src/arfs/input_validators.ts]: The regex is now split into three:
 - leading spaces
 - trailing dots and spaces
 - contains reserved characters
34a08f9

File: src/arfs/input_validators.ts (hunk 1 above)
Comment (matibat) [reply src/arfs/input_validators.ts]: There: 47690a7

File: src/arfs/arfs_entity_name_validators.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const RESERVED_CHARACTERS = ['\\\\', '/', ':', '*', '?', '"', '<', '>', '|'];
+const HUMAN_READABLE_RESERVED_CHARACTERS = RESERVED_CHARACTERS.map((char) => `'${char}'`).join(', ');
+
+const LEADING_SPACES_REGEXP = /^(\s+)/;
+const TRAILING_DOTS_OR_SPACES_REGEXP = /(\s|\.)+$/;
+const CONTAINS_RESERVED_CHARACTER_REGEXP = new RegExp(`[${RESERVED_CHARACTERS.join('')}]`, 'g');
+// From ArFS Standards
+const MAX_VALID_NAME_LENGTH = 255;
+
+export function assertValidArFSFileName(name: string): boolean {
```
Comment (arielmelendez) [inline src/arfs/arfs_entity_name_validators.ts:10]: The function looks great! Nit: it doesn't need a return value :)

File: src/arfs/arfs_entity_name_validators.test.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import { expect } from 'chai';
+import { assertValidArFSFileName } from './arfs_entity_name_validators';
+
+describe('entity name validators', () => {
```
Comment (arielmelendez) [inline src/arfs/arfs_entity_name_validators.test.ts:4]: Nice looking test file! Very readable!

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**